src/main/java/uk/gov/hmcts/reform/em/stitching/batch/DocumentTaskCallbackProcessor.java

```diff
@@ -50,38 +51,57 @@ public DocumentTaskCallbackProcessor(OkHttpClient okHttpClient, AuthTokenGenerat
 
     @Override
     public DocumentTask process(DocumentTask documentTask) throws InterruptedException {
+
+        Response response = null;
+        int callBackAttempts = documentTask.getCallback().getAttempts();
+
         try {
+            if (callBackAttempts < callBackMaxAttempts) {
```

Comment:
I would personally flip the logic here so that you're only checking the callBackAttempts in case of failure. that way you don't have to wait until the task gets picked in the next processing cycle to set it as failed.
you pick up the task, attempt the callback, if it fails, increment the count and check whether it reached the threshold or not. if it has, you fail it there and then, otherwise you update the callBackAttempts field